smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/StructureGenerator.kt

```diff
@@ -216,82 +202,49 @@ class StructureGenerator(
-            .withBlock("interface FluentBuilder {", "}") {
-                write("fun build(): #class.name:L")
+            .withBlock("class Builder {", "}") {
                 for (member in sortedMembers) {
                     val (memberName, memberSymbol) = memberNameSymbolIndex[member]!!
                     // we want the type names sans nullability (?) for arguments
                     writer.renderMemberDocumentation(model, member)
                     writer.renderAnnotations(member)
-                    write("fun #1L(#1L: #2T): FluentBuilder", memberName, memberSymbol)
-                }
-            }
-    }
-
-    private fun renderDslBuilderInterface() {
-        writer.write("")
-            .withBlock("interface DslBuilder {", "}") {
-                val structMembers: MutableList<MemberShape> = mutableListOf()
-
-                for (member in sortedMembers) {
-                    val (memberName, memberSymbol) = memberNameSymbolIndex[member]!!
-                    val targetShape = model.getShape(member.target).get()
-                    when {
-                        targetShape.isStructureShape -> structMembers.add(member)
-                    }
-
-                    writer.renderMemberDocumentation(model, member)
-                    writer.renderAnnotations(member)
-                    write("var #L: #P", memberName, memberSymbol)
-                }
-
-                write("")
-                write("fun build(): #class.name:L")
-                for (member in structMembers) {
-                    val (memberName, memberSymbol) = memberNameSymbolIndex[member]!!
-                    writer.dokka("construct an [${memberSymbol.fullName}] inside the given [block]")
-                    writer.renderAnnotations(member)
-                    openBlock("fun #L(block: #L.DslBuilder.() -> #Q) {", memberName, memberSymbol.name, KotlinTypes.Unit)
-                        .write("this.#L = #L.invoke(block)", memberName, memberSymbol.name)
-                        .closeBlock("}")
-                }
-            }
-    }
-
-    private fun renderBuilderImpl() {
-        writer.write("")
-            .withBlock("private class BuilderImpl() : FluentBuilder, DslBuilder {", "}") {
-                // override DSL properties
-                for (member in sortedMembers) {
-                    val (memberName, memberSymbol) = memberNameSymbolIndex[member]!!
-                    write("override var #L: #D", memberName, memberSymbol)
+                    write("var #L: #E", memberName, memberSymbol)
                 }
-
                 write("")
 
+                // generate the constructor used internally by serde
+                write("internal constructor()")
                 // generate the constructor that converts from the underlying immutable class to a builder instance
-                withBlock("constructor(x: #class.name:L) : this() {", "}") {
+                withBlock("constructor(x: #Q) : this() {", "}", symbol) {
```

Comment:
correctness

Should this be public?

Reply:
I'll make it internal and decorate with `@PublishedApi`.